### PR TITLE
Added type definitions for `burns`

### PR DESCRIPTION
**Pull request by tonyuki**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/burns/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+        "index.d.ts",
+        "burns-tests.ts"
+    ]
+}
```
Comment (kristianmitk) [inline types/burns/tsconfig.json]:
```suggestion
}

```

File: types/burns/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export interface Config {
+    defaultHandler?: any;
+    broadcaster?: any;
+    pusher?: any;
+}
+
+declare class Burns {
+    configure(config?: Config): this;
+
+    registerEvents(newEvents: ObjectLiteral): this;
```
Comment (rbuckton) [inline types/burns/index.d.ts]: TypeScript already provides a built-in `Record` type that satisfies this:

```suggestion
    registerEvents(newEvents: Record<string, any>): this;
```

File: types/burns/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// Type definitions for burns 2.0
+// Project: https://github.com/shalvah/burns#readme
+// Definitions by: Timothy Onyiuke <https://github.com/timolinn>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare const _Burns: Burns;
+
+export interface ObjectLiteral {
```
Comment (rbuckton) [inline types/burns/index.d.ts]: This isn't needed if you use the built-in `Record` type.

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/burns/index.d.ts (hunk 3 above)
Comment (timolinn) [reply types/burns/index.d.ts]: Fixed.

**Review by rbuckton — APPROVED** [review]

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**